### PR TITLE
Findbugs integration

### PR DESCRIPTION
**Pull request by oleg-nenashev**
@reviewbybees

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: pom.xml [hunk 1]
```diff
@@ -15,6 +15,38 @@
...
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <version>3.0.1</version>
+        <configuration>
+          <xmlOutput>true</xmlOutput>
+          <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
+          <failOnError>false</failOnError>
+        </configuration>
+        <executions>
```
Comment (jtnord) [inline pom.xml:33]: please configure an execution id.

**Comment by oleg-nenashev** [comment]
> please configure an execution id.

Do you have any preferences? I doubt it is required before we introduce the inheritance

**Comment by jtnord** [comment]
> Do you have any preferences?

just that it is descriptive

**Comment by jtnord** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/suppress_stack_trace/SuppressionFilter.java [hunk 2]
```diff
@@ -85,7 +86,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                     w.println(Messages.SuppressionFilter_Footer());
                     w.println("</body></html>");
                     w.close();
-                } catch (Exception x) {
+                } catch (Throwable x) {
```
Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/suppress_stack_trace/SuppressionFilter.java:91]: What happens if there is an `OutOfMemoryError`? you may not be able to create the `ServletException`

File: pom.xml (hunk 1 above)
Comment (oleg-nenashev) [inline pom.xml:33]: done

**Comment by oleg-nenashev** [comment]
@stephenc 
I've addressed your comment with catches

**Comment by stephenc** [comment]
:bee:

File: pom.xml [hunk 3]
```diff
@@ -15,6 +15,44 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <version>${findbugs-maven-plugin.version}</version>
+        <configuration>
+          <xmlOutput>true</xmlOutput>
+          <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
+          <failOnError>${findbugs.failOnError}</failOnError>
```
Comment (jtnord) [inline pom.xml:31]: this is the default - no need to waste a line of XML.

File: pom.xml [hunk 4]
```diff
@@ -15,6 +15,44 @@
...
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <version>${findbugs-maven-plugin.version}</version>
+        <configuration>
+          <xmlOutput>true</xmlOutput>
+          <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
```
Comment (jtnord) [inline pom.xml]: as others pointed out - this doesn't exist.

File: pom.xml (hunk 3 above)
Comment (oleg-nenashev) [inline pom.xml:31]: I would keep it explicit just to allow searching for usages of the property

**Comment by oleg-nenashev** [comment]
@jtnord Done

**Comment by jtnord** [comment]
:bee:

**Pull request merged**